src/util/hash.ts

```diff
@@ -47,8 +47,8 @@ function nextPowerOf2 (n: number): number {
 }
 
 /** @ignore */
-export function merkleize(chunks: Buffer[]): Buffer {
-  const lengthToPad = nextPowerOf2(chunks.length) - chunks.length;
+export function merkleize(chunks: Buffer[], padFor: number = 0): Buffer {
+  const lengthToPad = nextPowerOf2(padFor || chunks.length) - chunks.length;
   if (lengthToPad) {
     // Add zeroed chunks as leaf nodes to create full binary tree
     const emptyChunk = Buffer.alloc(BYTES_PER_CHUNK);
```

Comment:
Unrelated to PR, why not `Buffer.alloc(BYTES_PER_CHUNK * lengthToPad)`. I'm asking because this method(merkleize) is kinda a bottleneck

Reply:
its definitely possible to optimize this function. I'm thinking about storing zero-hashes like in our lodestar merkle tree implementation.
Here we're creating empty "chunks" that we use to hash, merkle-style.
We can't allocate a chunk * lengthToPad because we need to hash each chunk with each next chunk.

Reply:
yeah but you are doing same thing in line below but this would at least remove concatenation in line below

Reply:
its slightly different, its an Array of Buffers, not one big Buffer. Each element in the array is a leaf in the merkle tree.

Reply:
damm, I'm stupid, whole time I thought it's `Buffer.concat` but it's `chunks.concat` sorry